hs_core/models.py

```diff
@@ -1354,6 +1303,77 @@ class AbstractResource(ResourcePermissionsMixin):
         class MyResourceContentType(pages.Page, hs_core.AbstractResource):
             ...
     """
+    # For each request for a resource page view, the Page will run can_view(), can_change(),
+    # can_delete() in succession to check for permissions, so for performance improvement,
+    # there might be merit to make ha_obj a class member variable with a persistent connection
+    # for the request, rather than a separate connection within each method call
+    def can_add(self, request):
+        return self.can_change(request)
+
+    def can_delete(self, request):
+        user = get_user(request)
+        if user.is_authenticated():
+            if user.is_superuser:
+                ret = True
+            elif self.creator and user.pk == self.creator.pk:
```

Comment:
@ironfroggy is correct except that when he referred to creator that is the "Django object creator" not the "intellectual content creator".  @alvacouch also indicated in his notes on the access control document that the way ceding ownership is implemented requires specifying another privilege level for self.